coral-component-colorpicker/src/scripts/ColorArea.js

```diff
@@ -0,0 +1,560 @@
+  /** @private **/
+  _updateHue(hue) {
+    this._hue = hue;
+    if(this._gradient){
+      this._gradient.style.background = `linear-gradient(to top, black 0%, hsla(${this._hue}, 100%, 0%, 0) 100%),
+        linear-gradient(to right, white 0%, hsla(${this._hue}, 100%, 0%, 0) 100%),
+        hsl(${this._hue}, 100%, 50%)`;
+    }  
+  }
+        
+  /** @private */
+  _syncInputSliderAttrs() {
+    if(Number(this._elements.sliderX.getAttribute('min')) !== this._minX()) {
+      this._elements.slideX.setAttribute('min', this._minX());
+    }
+    
+    if(Number(this._elements.sliderX.getAttribute('max')) !== this._maxX()) {
+      this._elements.slideX.setAttribute('max', this._maxX());
+    }
+    
+    if(Number(this._elements.sliderX.getAttribute('step')) !== this._stepX()) {
+      this._elements.slideX.setAttribute('step', this._stepX());
+    }  
+
+    if(Number(this._elements.sliderY.getAttribute('min')) !== this._minY()) {
+      this._elements.sliderY.setAttribute('min', this._minY());
+    }
+    
+    if(Number(this._elements.sliderY.getAttribute('max')) !== this._maxY()) {
+      this._elements.sliderY.setAttribute('max', this._maxY());
+    }
+    
+    if(Number(this._elements.sliderY.getAttribute('step')) !== this._stepY()) {
+      this._elements.sliderY.setAttribute('step', this._stepY());
+    } 
+        
+    if(this._elements.sliderX.getAttribute('aria-label') !== this._label) {
+      this._elements.sliderX.setAttribute('aria-label', this._label);
+      this._elements.sliderY.setAttribute('aria-label', this._label);
+    }
+  }
+
+  /** @private */  
+  _snapValueToStep(rawValue, min, max, step) {
+    const remainder = (rawValue - min) % step;
+    let snappedValue = rawValue;
+    
+    if (Math.abs(remainder) * 2 >= step) {
+      snappedValue = rawValue - remainder + step;
+    } 
+    else {
+      snappedValue = rawValue - remainder;
+    }
+    if (snappedValue < min) {
+      snappedValue = min;
+    } 
+    else if (snappedValue > max) {
+      snappedValue = min + Math.floor((max - min) / step) * step;
+    }
+    return snappedValue;
+  }
+  
+  /** @private */
+  _updateHandle(hue, x, y, colorStr) {
+    let percent = 100 - ((y - this._minY()) / (this._maxY() - this._minY()) * 100);
+    if(this._handle) {
+      this._handle.style.top = `${percent}%`;
+    }
+    
+    percent = (x - this._minX()) / (this._maxX() - this._minX()) * 100;
+    if(this._handle) {
+      this._handle.style.left = `${percent}%`;
+    }   
+    this._elements.colorHandle.setAttribute('color', colorStr); 
+  }
+    
+  /** @private */  
+  _focusHandle(isFocused) {
+    if(this._handle) {
+       if(isFocused === true) {
+        this._handle.focus();  
+      } 
+      else {
+        this._handle.blur(); 
+      }
+    }
+  }
+
+  /** @private */ 
+  _changeValue(x, y) {
+     if(this.x !== x) {
+       this.x = x;
+       this.trigger('change');
+     }
+     if(this.y !== y) {
+       this.y = y;
+       this.trigger('change');
```

Comment:
We will be calling a change event twice in case both value change.  If possible can we trigger only one change event.